contrib/containers/guix/scripts/guix-start

```diff
@@ -25,8 +25,12 @@ if [[ ! -d "${SDK_PATH}/${XCODE_ARCHIVE}" ]]; then
     curl -L https://bitcoincore.org/depends-sources/sdks/${XCODE_ARCHIVE}.tar.gz | tar -xz -C "${SDK_PATH}"
 fi
 
+# Add safe.directory option only if it's not present yet
+if [[ -z `git config --global --get-all safe.directory | grep -Fx "${WORKSPACE_PATH}"` ]]; then
```

Comment:
nit: always prefer $() over `` for bash scripts. It works with "", it can be used recursively. \`\` has no benefits except "¹ character shorter"

Btw, why safe.directory can't be set twice?

Reply:
Without this `if` your git config will start looking like this after some time
```
[safe]
        directory = /root/dash
        directory = /root/dash
...
        directory = /root/dash
        directory = /root/dash
```
Adjust the comment to hopefully explain it better :)

Reply:
thanks for explanation how it works.

_I thought for a moment, that it's local .gitconfig is subject to be changed by this way _sweaty_face_ Because it is so malicious behaviour to update .gitconfig without asking user._

so, it means to be called inside Docker container only, right?

I'd propose this way to set safe directory then, it exclude calling external command `grep` (can be done with IF or with ||, I am good both ways):
```
git config --global --fixed-value --get safe.directory "${WORKSPACE_PATH}$" || git config --global --add safe.directory "${WORKSPACE_PATH}"
```

Reply:
>"${WORKSPACE_PATH}$"

Is extra `$` at the end of it just a typo or is it intentional?

Reply:
Typo, result of editing after copy paste